packages/test/src/test-custom-data-converter.ts

```diff
@@ -0,0 +1,143 @@
+  }
+  t.deepEqual(
+    new coresdk.activity_result.ActivityResult(actual ?? undefined).toJSON(),
+    new coresdk.activity_result.ActivityResult(expected).toJSON()
+  );
+}
+
+if (RUN_INTEGRATION_TESTS) {
+  test('Client and Worker work with provided dataConverter/dataConverterPath', async (t) => {
+    let resolvePromise: (value: unknown) => void;
+    const receivedExpectedError = new Promise(function (resolve) {
+      resolvePromise = resolve;
+    });
+
+    await Core.install({
+      logger: new DefaultLogger('ERROR', (entry) => {
+        if (
+          entry.message === 'Failed to activate workflow' &&
+          entry.meta?.error?.stack?.includes('Activator.startWorkflow') &&
+          entry.meta?.error?.message ===
+            'Unable to deserialize protobuf message without protobufClasses provided to DefaultDataConverter'
+        ) {
+          resolvePromise(true);
+        }
+      }),
+    });
+
+    const taskQueue = 'custom-data-converter';
+    const worker = await Worker.create({
+      ...defaultOptions,
+      taskQueue,
+      dataConverterPath: require.resolve('./data-converters/data-converter'),
+    });
+    const connection = new Connection();
+    const client = new WorkflowClient(connection.service, { dataConverter });
+
+    // For now, just check that the protobuf message gets to the workflow
+    worker.run();
+    client.execute(protobufWorkflow, {
+      args: [messageInstance],
+      workflowId: uuid4(),
+      taskQueue,
+    });
+    await receivedExpectedError;
+    worker.shutdown();
+    t.pass();
+
+    // const runAndShutdown = async () => {
+    //   const result = await client.execute(protobufWorkflow, {
+    //     args: [messageInstance],
+    //     workflowId: uuid4(),
+    //     taskQueue,
+    //   });
+    //   t.is(result, messageInstance);
+    //   worker.shutdown();
+    // };
+    // await Promise.all([worker.run(), runAndShutdown()]);
+  });
+}
+
+test('Worker throws on invalid dataConverterPath', async (t) => {
+  await t.throwsAsync(
+    isolateFreeWorker({
+      ...defaultOptions,
+      dataConverterPath: './wrong-path',
+    }),
+    {
+      message: /Could not find a file at the specified dataConverterPath/,
+    }
+  );
+
+  await t.throwsAsync(
+    isolateFreeWorker({
+      ...defaultOptions,
+      dataConverterPath: require.resolve('./data-converters/data-converter-no-export'),
+    }),
+    {
+      message: /The module at dataConverterPath .* does not have a `dataConverter` named export./,
+    }
+  );
+
+  await t.throwsAsync(
+    isolateFreeWorker({
+      ...defaultOptions,
+      dataConverterPath: require.resolve('./data-converters/data-converter-bad-export'),
+    }),
+    {
+      message:
+        /The `dataConverter` named export at dataConverterPath .* should be an instance of a class that implements the DataConverter interface./,
+    }
+  );
+});
+
+test('Worker with proto data converter runs an activity and reports completion', async (t) => {
```

Comment:
Might as well have made this a full integration test where the workflow calls the activity

Reply:
I was trying to skip the workflow in this test since it can't receive protobufs yet. But I guess the workflow could create a protobuf just to use when calling the activity. If that's helpful I could add/replace?